src/main/java/de/uhd/ifi/se/decision/management/jira/quality/MetricCalculator.java

```diff
@@ -52,343 +53,365 @@
-			if (gitClient != null) {
-				for (Issue jiraIssue : jiraIssues) {
-					int numberOfCommits = gitClient.getNumberOfCommits(jiraIssue);
-					resultMap.put(jiraIssue.getKey(), numberOfCommits);
-				}
-			}
-		} catch (Exception ex) {
-			LOGGER.error(ex.getMessage());
-		}
-		return resultMap;
-	}
-
-	public Map<String, Integer> getNumberOfDecisionKnowledgeElementsForJiraIssues(KnowledgeType type,
-			Integer linkDistance) {
-		Map<String, Integer> numberOfSentencesPerIssue = new HashMap<String, Integer>();
-		for (Issue jiraIssue : jiraIssues) {
-			int numberOfElements = 0;
-			Map<Integer, List<Issue>> linkDistanceMap = getLinkDistanceIssueMap(linkDistance, jiraIssue);
-			for (int i = 0; i <= linkDistance; i++) {
-				for (Issue issue : linkDistanceMap.get(i)) {
-					List<KnowledgeElement> elements = KnowledgePersistenceManager.getOrCreate(projectKey)
-							.getJiraIssueTextManager().getElementsInJiraIssue(issue.getId());
-					if (issue.getIssueType().getName().equals(type.toString())) {
-						numberOfElements++;
-					}
-					for (KnowledgeElement element : elements) {
-						if (element.getType().equals(type)) {
-							numberOfElements++;
-						}
-					}
-					if (i <= linkDistance - 2 && decisionKnowledgeCodeElements != null) {
-						for (KnowledgeElement element : decisionKnowledgeCodeElements) {
-							if (element.getType().equals(type)) {
-								numberOfElements++;
-							}
-						}
-					}
-					if (i > 0 && i <= (linkDistance - 1) && decisionKnowledgeCommitElements != null) {
-						for (KnowledgeElement element : decisionKnowledgeCommitElements) {
-							if (element.getType().equals(type)) {
-								numberOfElements++;
-							}
-						}
-					}
-				}
-			}
-			numberOfSentencesPerIssue.put(jiraIssue.getKey(), numberOfElements);
-		}
-		return numberOfSentencesPerIssue;
-	}
+    /*
+     * public Map<String, Integer> numberOfCommitsPerIssue() { Map<String, Integer>
+     * resultMap = new HashMap<String, Integer>(); if (resultMap.size() == 0) {
+     * return null; } try { if (this.gitClient != null) { for (Issue jiraIssue :
+     * jiraIssues) { int numberOfCommits =
+     * this.gitClient.getNumberOfCommits(jiraIssue);
+     * resultMap.put(jiraIssue.getKey(), numberOfCommits); } } } catch (Exception
+     * ex) { LOGGER.error(ex.getMessage()); } System.out.println("5"); return
+     * resultMap; }
+     */
 
-	public Map<String, Integer> getDistributionOfKnowledgeTypes() {
-		Map<String, Integer> distributionOfKnowledgeTypes = new HashMap<String, Integer>();
-		for (KnowledgeType type : KnowledgeType.getDefaultTypes()) {
-			int numberOfElements = graph.getElements(type).size();
-			for (KnowledgeElement element : (Optional.ofNullable(decisionKnowledgeCodeElements)
-					.orElse(Collections.emptyList()))) {
-				if (element.getType().equals(type)) {
-					numberOfElements++;
-				}
+    public Map<String, Integer> getNumberOfDecisionKnowledgeElementsForJiraIssues(KnowledgeType type,

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getNumberOfDecisionKnowledgeElementsForJiraIssues(KnowledgeType, Integer)' has an NPath complexity of 246](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=4909482)